test/e2e/create_test.go

```diff
@@ -175,17 +175,68 @@ func (tc *testContext) testMachineConfiguration(t *testing.T) {
 	}
 	_, err := tc.createWindowsMachineSet(gc.numberOfMachineNodes, false)
 	require.NoError(t, err, "failed to create Windows MachineSet")
+	machineCreationTime := time.Now()
 
 	t.Run("Machine configuration while private key change", tc.testMachineConfigurationWhilePrivateKeyChange)
 
 	machines, err := tc.waitForWindowsMachines(int(gc.numberOfMachineNodes), "Provisioned", false)
 	require.NoError(t, err, "error waiting for Windows Machines to be provisioned")
+	machineProvisionedTime := time.Now()
+	var addresses []string
+	for _, m := range machines.Items {
+		addr, err := controllers.GetAddress(m.Status.Addresses)
+		require.NoError(t, err)
+		addresses = append(addresses, addr)
+	}
+	err = tc.waitForSSHAvailable(addresses)
+	require.NoError(t, err)
+	sshAvailableTime := time.Now()
 	err = tc.waitForConfiguredWindowsNodes(gc.numberOfMachineNodes, false, false)
 	assert.NoError(t, err, "Windows node creation failed")
+
+	// Save timings as an artifact for performance measuring purposes
+	configTimes := configurationTimes{
+		NodeType:           "machine",
```

Comment:
I recommend using `InstanceType` instead, NodeType may be confused with the worker/control-plane standard mapping.